serde/src/ser/mod.rs

```diff
@@ -616,6 +620,29 @@ pub trait Serializer: Sized {
         }
         serializer.end()
     }
+
+    /// Collect a `Display` value as a string.
+    ///
+    /// The default implementation serializes the given value as a string with
+    /// `ToString::to_string`.
+    #[cfg(any(feature = "std", feature = "collections"))]
+    fn collect_str<T>(self, value: &T) -> Result<Self::Ok, Self::Error>
+        where T: Display,
+    {
+        let mut string = String::new();
+        write!(string, "{}", value).unwrap();
```

Comment:
Any advantages of this over `value.to_string()`? I guess in case there is a specialized ToString impl to make this default implementation behave consistently with a serializer that overrides collect_str with a more efficient Display-based implementation? I think people deserve what they get if they implement ToString and Display in a way that is not consistent with each other.

ToString may be faster https://github.com/rust-lang/rust/issues/18404 (not that I expect this method to be used with T=str).